### DISCUSSION TITLE
Insert description in a option on ui.select

### DISCUSSION THREAD

**Discussion by ChristianKJ** [zauberzeug/nicegui#1987] [Ideas / Feature Requests]
### Description

I want to put the description of an option, I saw in the quasar documentation that I can insert an array of objects into the options, but as soon as I try to insert it into ui.select I get the following output in the select [ object object], how do I set a description?

My code:
```py
with ui.card_section():
    ui.select(options=[
        {"label": "NameLabel1", "value": 1, "description": "descriptionTest1"},
        {"label": "NameLabel2", "value": 2, "description": "descriptionTest2"}
    ], label="TestSelect").props('outlined')
```

My output frontend:
<img width="475" alt="image" src="https://github.com/zauberzeug/nicegui/assets/95303392/f70f0f50-bd3c-40eb-a6ce-debe7937eb3b">

Documentation Quasar
https://quasar.dev/vue-components/select#customizing-menu-options

**Comment by falkoschindler** [comment]
Unfortunately, it is currently impossible to add additional attributes to options in `ui.select`. The `options` list or dictionary is not directly passed to Quasar, but transformed with indices:
https://github.com/zauberzeug/nicegui/blob/efe275585a499367534ee98b4bb0a7066ff0bf4b/nicegui/elements/choice_element.py#L27

Maybe we can add this feature in the future. It's just a little tricky to meet all the different requirements of `ui.select`, e.g. additional options via free-text input, multi-select etc. But who knows.

**Reply by Rubp-007** [reply]
@falkoschindler It'll be nice to have such a feature that we can set options of `ui.select` as a list of dict, just like `ui.tree` :)

I wonder if we have some "hacks" for now.

**Reply by Rubp-007** [reply]
Oh I found the one you posted here: https://github.com/zauberzeug/nicegui/discussions/1694#discussioncomment-7114752. It works great!

**Reply by falkoschindler** [reply]
Oh right! I forgot about this approach. 😄
